src/monitor/node_metadata.c

```diff
@@ -360,7 +360,12 @@ FindFailoverNewStandbyNode(List *groupNodeList)
  *	  qsort comparator for sorting node lists by candidate priority
  */
 static int
-pgautofailover_node_candidate_priority_compare(const void *a, const void *b)
+#if (PG_VERSION_NUM >= 130000)
+pgautofailover_node_candidate_priority_compare(const union ListCell *a,
+											   const union ListCell *b)
+#else
+pgautofailover_node_candidate_priority_compare(const void * a, const void * b)
+#endif
 {
 	AutoFailoverNode *node1 = (AutoFailoverNode *) lfirst(*(ListCell **) a);
 	AutoFailoverNode *node2 = (AutoFailoverNode *) lfirst(*(ListCell **) b);
```

Comment:
I'm surprised this works both in Postgres 12 and Postgres 13, where I read the function prototype in a way that we are now doing one pointer dereference too many. I am not sure that we exercise this code path yet in the test suite, it's part of the Multiple Standby PR building blocks... which might explain that it could compile but it's going to fail.

Reply:
Fixed now.